src/components/LocalIdentityModal/LocalIdentityModal.js

```diff
@@ -7,14 +7,26 @@ import EscapeOutside from '../EscapeOutside/EscapeOutside'
 import IdentityBadgeWithNetwork from '../IdentityBadge/IdentityBadgeWithNetwork'
 import keycodes from '../../keycodes'
 
-const LocalIdentityModal = ({ opened, ...props }) => {
-  const { showModal, hideModal } = React.useContext(ModalContext)
-  React.useEffect(() => {
-    opened ? showModal(Modal, props) : hideModal()
-  }, [opened, showModal, hideModal, props])
+const LocalIdentityModal = React.memo(
+  ({ opened, address, label, onCancel, onSave }) => {
+    const { showModal, hideModal } = React.useContext(ModalContext)
+
+    const modalProps = React.useMemo(
+      () => ({ address, label, onCancel, onSave }),
+      [address, label, onCancel, onSave]
+    )
+
+    React.useEffect(() => {
+      if (opened) {
+        showModal(Modal, modalProps)
+      } else {
+        hideModal()
+      }
+    }, [opened, modalProps, showModal, hideModal])
```

Comment:
Is there a difference if we spread the modalProps here? ` }, [opened, ...modalProps, showModal, hideModal])`

Reply:
No we could have done that too, ideally listing them explicitly rather than using the spread operator, so that the `react-hooks` ESLint plugin can get enough information. I only separated these for readability, so that these values don’t get mixed with the others in the dependency list.